File: SK210 - Flutter/android/app/src/main/java/com/example/DV/sk210flutter/sensor_bobina/SensorBobina.java
package com.example.DV.sk210flutter.sensor_bobina;

import android.os.Bundle;
import android.os.RemoteException;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.DV.sk210flutter.DeviceServiceManager;
import com.example.DV.sk210flutter.R;

public class SensorBobina extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sensor_bobina);

        Button redButton = findViewById(R.id.redButton);
        redButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                // Verificar o estado do papel
                int paperState = checkPaperState();

                // Verificar se a tampa da impressora está aberta
                boolean isCoverOpen = checkCoverState();

                // Verificar o status da impressora
                int printerState = checkPrinterState();
                if (paperState == 0) {
                    Toast.makeText(SensorBobina.this, "O papel está quase acabando", Toast.LENGTH_SHORT).show();

                } else if (printerState == 0x01) {
                    // A impressora está sem papel, acender o LED vermelho

                    Toast.makeText(SensorBobina.this, "O papel acabou", Toast.LENGTH_SHORT).show();

                } else if (printerState == 0x00 && !isCoverOpen) {

                    Toast.makeText(SensorBobina.this, "Há papel suficiente", Toast.LENGTH_SHORT).show();

                }
            }
        });
    }

    // Método para verificar o estado do papel
    private int checkPaperState() {
        try {
            return DeviceServiceManager.getInstance().getPrintManager().getPosPrintPaperState();
        } catch (RemoteException e) {
            throw new RuntimeException(e);
        }
    }

    // Método para verificar se a tampa da impressora está aberta
    private boolean checkCoverState() {
        try {
            return DeviceServiceManager.getInstance().getPrintManager().isPosPrinterCoverOpen();
        } catch (RemoteException e) {
            throw new RuntimeException(e);
        }
    }

    // Método para verificar o status da impressora
    private int checkPrinterState() {
        try {
            return DeviceServiceManager.getInstance().getPrintManager().getPrinterState();
        } catch (RemoteException e) {
            throw new RuntimeException(e);
        }
    }
}

File: SK210 - Kotlin/app/src/main/java/com/exampledv/exemplosk210kot/sensor/SensorBobina.kt
package com.exampledv.exemplosk210kot.sensor

import android.os.Bundle
import android.os.RemoteException
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.exampledv.exemplosk210kot.R
import com.exampledv.exemplosk210kot.util.DeviceServiceManager

class SensorBobina : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sensor_bobina)

        val redButton = findViewById<Button>(R.id.redButton)
        redButton.setOnClickListener {
            // Verificar o estado do papel
            val paperState = checkPaperState()

            // Verificar se a tampa da impressora está aberta
            val isCoverOpen = checkCoverState()

            // Verificar o status da impressora
            val printerState = checkPrinterState()
            if (paperState == 0) {
                Toast.makeText(
                    this@SensorBobina,
                    "O papel está quase acabando",
                    Toast.LENGTH_SHORT
                ).show()
            } else if (printerState == 0x01) {
                // A impressora está sem papel, acender o LED vermelho

                Toast.makeText(
                    this@SensorBobina,
                    "O papel acabou",
                    Toast.LENGTH_SHORT
                ).show()
            } else if (printerState == 0x00 && !isCoverOpen) {
                Toast.makeText(
                    this@SensorBobina,
                    "Há papel suficiente",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }

    // Método para verificar o estado do papel
    private fun checkPaperState(): Int {
        try {
            return DeviceServiceManager.getInstance().getPrintManager().getPosPrintPaperState()
        } catch (e: RemoteException) {
            throw RuntimeException(e)
        }
    }

    // Método para verificar se a tampa da impressora está aberta
    private fun checkCoverState(): Boolean {
        try {
            return DeviceServiceManager.getInstance().getPrintManager().isPosPrinterCoverOpen()
        } catch (e: RemoteException) {
            throw RuntimeException(e)
        }
    }

    // Método para verificar o status da impressora
    private fun checkPrinterState(): Int {
        try {
            return DeviceServiceManager.getInstance().getPrintManager().getPrinterState()
        } catch (e: RemoteException) {
            throw RuntimeException(e)
        }
    }
}